src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs

```diff
@@ -13,18 +13,27 @@ internal static class ProjectExtensions
     {
         public static ProjectType GetProjectType(this Project project)
         {
-            var projectType = ProjectType.Console;
             if (project.IsWebProject())
             {
-                projectType = ProjectType.Web;
+                return ProjectType.Web;
+            }
+            
+            if (project.IsLibrary())
+            {
+                return ProjectType.Library;
             }
 
-            return projectType;
+            return ProjectType.Console;
+        }
+
+        private static bool IsLibrary(this Project project)
+        {
+            return !project.IsTestProject && !project.GetCompilerOptions(null, "Debug").EmitEntryPoint.GetValueOrDefault();
```

Comment:
We are now duplicating this code to generate the compiler options. Can we do that in the top method and pass it down?